src/crl.rs

```diff
@@ -0,0 +1,150 @@
+    /// sign this CRL.
+    pub authority_key_identifier: Option<untrusted::Input<'a>>,
+
+    /// A monotonically increasing sequence number for a given CRL scope and CRL issuer.
+    pub crl_number: Option<&'a [u8]>,
+}
+
+impl<'a> core::convert::TryFrom<&'a [u8]> for CertRevocationList<'a> {
+    type Error = Error;
+
+    /// Try to parse the given bytes as a RFC 5280[^1] profile Certificate Revocation List (CRL).
+    ///
+    /// Webpki does not support:
+    ///   * CRL versions other than version 2.
+    ///   * CRLs missing the next update field.
+    ///   * CRLs missing certificate revocation list extensions.
+    ///   * Delta CRLs.
+    ///
+    /// [^1]: <https://www.rfc-editor.org/rfc/rfc5280#section-5>
+    fn try_from(crl_der: &'a [u8]) -> Result<Self, Self::Error> {
+        parse_crl(untrusted::Input::from(crl_der))
+    }
+}
+
+fn parse_crl(crl_der: untrusted::Input) -> Result<CertRevocationList, Error> {
+    let (tbs_cert_list, signed_data) = crl_der.read_all(Error::BadDer, |crl_der| {
+        der::nested(
+            crl_der,
+            Tag::Sequence,
+            Error::BadDer,
+            signed_data::parse_signed_data,
+        )
+    })?;
+
+    tbs_cert_list.read_all(Error::BadDer, |tbs_cert_list| {
+        version2(tbs_cert_list)?;
+
+        // RFC 5280 §5.1.2.2:
+        //   This field MUST contain the same algorithm identifier as the
+        //   signatureAlgorithm field in the sequence CertificateList
+        let signature = der::expect_tag_and_get_value(tbs_cert_list, Tag::Sequence)?;
+        if signature != signed_data.algorithm {
+            return Err(Error::SignatureAlgorithmMismatch);
+        }
+
+        // RFC 5280 §5.1.2.3:
+        //   The issuer field MUST contain a non-empty X.500 distinguished name (DN).
+        let issuer = der::expect_tag_and_get_value(tbs_cert_list, Tag::Sequence)?;
+
+        // RFC 5280 §5.1.2.4:
+        //    This field indicates the issue date of this CRL.  thisUpdate may be
+        //    encoded as UTCTime or GeneralizedTime.
+        // We do not presently enforce the correct choice of UTCTime or GeneralizedTime based on
+        // whether the date is post 2050.
+        let this_update = der::time_choice(tbs_cert_list)?;
+
+        // While OPTIONAL in the ASN.1 module, RFC 5280 §5.1.2.5 says:
+        //   Conforming CRL issuers MUST include the nextUpdate field in all CRLs.
+        // We do not presently enforce the correct choice of UTCTime or GeneralizedTime based on
+        // whether the date is post 2050.
+        let next_update = der::time_choice(tbs_cert_list)?;
+
+        // RFC 5280 §5.1.2.6:
+        //   When there are no revoked certificates, the revoked certificates list
+        //   MUST be absent
+        // TODO(@cpu): Do we care to support empty CRLs if we don't support delta CRLs?
+        let revoked_certs = if tbs_cert_list.peek(Tag::Sequence.into()) {
```

Comment:
Maybe `revoked_certs` could be an empty `untrusted::Input` instead of wrapping it in `Option` in the case it ends up being absent?